pkg/controllers/clusterset/clusterclaim/clusterclaim_controller.go

```diff
@@ -54,11 +54,47 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	if err = c.Watch(&source.Kind{Type: &hivev1.ClusterClaim{}},
-		&handler.EnqueueRequestForObject{}); err != nil {
+	err = c.Watch(&source.Kind{Type: &hivev1.ClusterClaim{}},
+		&handler.EnqueueRequestForObject{})
+	if err != nil {
 		return err
 	}
 
+	//watch clusterdeployment related claim
+	err = c.Watch(
+		&source.Kind{Type: &hivev1.ClusterDeployment{}},
+		handler.EnqueueRequestsFromMapFunc(
+			handler.MapFunc(func(a client.Object) []reconcile.Request {
+				clusterDeployment, ok := a.(*hivev1.ClusterDeployment)
+				if !ok {
+					// not a ClusterDeployment, returning empty
+					klog.Error("ClusterDeployment handler received non-ClusterDeployment object")
+					return []reconcile.Request{}
+				}
+				clusterclaims := &hivev1.ClusterClaimList{}
+				err := mgr.GetClient().List(context.TODO(), clusterclaims, &client.ListOptions{})
+				if err != nil {
+					klog.Errorf("could not list clusterclaims. Error: %v", err)
+				}
+				var requests []reconcile.Request
+				for _, clusterclaim := range clusterclaims.Items {
+					//If clusterclaims is not related to this clusterDeployment, ignore it
+					if clusterclaim.Spec.Namespace != clusterDeployment.Namespace {
+						continue
+					}
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name:      clusterclaim.Spec.Namespace,
```

Comment:
name or namespace?

Reply:
it should be namespace, it means clusterdeployment namespace(same as name)